xarray/tests/test_array_api.py

```diff
@@ -137,6 +137,9 @@ def test_unstack(arrays: tuple[xr.DataArray, xr.DataArray]) -> None:
     assert_equal(actual, expected)
 
 
+@pytest.mark.xfail(
+    reason="We don't know how to handle python scalars with array API arrays. This is strictly prohibited by the array API"
+)
```

Comment:
the `xfail` is still necessary, because we're running `xp.result_type` on all inputs one-by-one, including the scalars. In theory, we could replace that by a `npcompat` wrapper that dispatches to either `numpy.result_type` or `_fallback_array_api_result_type`.

Reply:
Move this comment into the code?

Reply:
I decided to move the dispatching to `npcompat` instead and extend `_future_array_api_result_type` to support passing only weakly-dtyped values.